orbit/cmd/orbit/orbit.go

```diff
@@ -516,6 +518,36 @@ func main() {
 		}
 		g.Add(flagRunner.Execute, flagRunner.Interrupt)
 
+		// only setup extensions autoupdate if we have enabled updates
+		if !c.Bool("disable-updates") || c.Bool("dev-mode") {
+			const orbitExtensionUpdateInterval = 60 * time.Second
+			extRunner := update.NewExtensionConfigUpdateRunner(orbitClient, update.ExtensionUpdateOptions{
+				CheckInterval: orbitExtensionUpdateInterval,
+				RootDir:       c.String("root-dir"),
+			}, updateRunner)
+
+			if _, err := extRunner.DoExtensionConfigUpdate(); err != nil {
+				// just log, OK to continue since this will get retry
+				log.Info().Err(err).Msg("initial update to fetch extensions from /config API failed")
+			}
+
+			// call UpdateAction on the updateRunner after we have fetched extensions from Fleet
+			_, err := updateRunner.UpdateAction()
+			if err != nil {
+				// OK, initial call may fail, ok to continue
+				log.Debug().Err(err).Msg("extensions update action failed")
+			}
+
+			extensionAutoLoadFile := filepath.Join(c.String("root-dir"), "extensions.load")
+			stat, err := os.Stat(extensionAutoLoadFile)
+			// we only want to add the extensions_autoload flag to osquery, if the file exists and size > 0
```

Comment:
What happens if:
1. Initially there's no `extensions_autoload`, and `osqueryd` is started without it
2. Thereafter the user pushes and configures extensions
3. Will Orbit restart and run this part again? (to run `osqueryd` with the `--extensions_autoload` flag)

Reply:
Yes, orbit will restart and run this again. I have specifically tested this case.